docs/workflow/building/libraries/cross-building.md

```diff
@@ -24,108 +23,102 @@ and for arm64 you need:
     $ sudo apt-get install binutils-aarch64-linux-gnu
 
 
-Generating the rootfs
+Generate the rootfs
 ---------------------
-The `eng/common/cross/build-rootfs.sh` script can be used to download the files needed for cross compilation. It will generate an Ubuntu 16.04 rootfs as this is what CoreFX targets.
+The `eng/common/cross/build-rootfs.sh` script can be used to download the files needed for cross compilation. It can generate rootfs for different operating systems.
 
-    Usage: ./eng/common/cross/build-rootfs.sh [BuildArch] [LinuxCodeName] [lldbx.y] [--skipunmount] --rootfsdir <directory>]
-    BuildArch can be: arm, armel, arm64, x86
-    LinuxCodeName - optional, Code name for Linux, can be: trusty, xenial(default), zesty, bionic, alpine. If BuildArch is armel, LinuxCodeName is jessie(default) or tizen.
-    lldbx.y - optional, LLDB version, can be: lldb3.9(default), lldb4.0, lldb5.0, lldb6.0 no-lldb. Ignored for alpine
+    Usage: ./eng/common/cross/build-rootfs.sh [BuildArch] [CodeName] [lldbx.y] [llvmx[.y]] [--skipunmount] --rootfsdir <directory>]
+    BuildArch can be: arm(default), arm64, armel, armv6, ppc64le, riscv64, s390x, x64, x86
+    CodeName - optional, Code name for Linux, can be: xenial(default), zesty, bionic, alpine, alpine3.13 or alpine3.14. If BuildArch is armel, LinuxCodeName is jessie(default) or tizen.
+                                   for FreeBSD can be: freebsd12, freebsd13
+                                   for illumos can be: illumos
+                                   for Haiku can be: haiku
 
-The `build-rootfs.sh` script must be run as root, as it has to make some symlinks to the system. It will, by default, generate the rootfs in `cross/rootfs/<BuildArch>` however this can be changed by setting the `ROOTFS_DIR` environment variable or by using --rootfsdir.
+The `build-rootfs.sh` script must be run as root, as it has to make some symlinks to the system. By default it generates the rootfs in `.tools/rootfs/<BuildArch>`, however this can be changed by setting the `ROOTFS_DIR` environment variable or by using `--rootfsdir`.
 
-For example, to generate an arm rootfs:
+For example, to generate an arm Ubuntu 18.04 rootfs:
 
-    $ sudo ./eng/common/cross/build-rootfs.sh arm
+    $ sudo ./eng/common/cross/build-rootfs.sh arm bionic
 
-and if you wanted to generate the rootfs elsewhere:
+And to generate the rootfs elsewhere:
 
-    $ sudo ./build-rootfs.sh arm --rootfsdir  /mnt/corefx-cross/arm
+    $ sudo ./build-rootfs.sh arm bionic --rootfsdir /mnt/rootfs/arm
 
 
-Cross compiling for native CoreFX
+Compile native part of libraries
 ---------------------------------
-Once the rootfs has been generated, it will be possible to cross compile CoreFX. If `ROOTFS_DIR` was set when generating the rootfs, then it must also be set when running `build.sh`.
 
-So, without `ROOTFS_DIR`:
+To build native part of libraries for arm using subset:
```

Comment:
Same nit: `libs.native` or `native runtime libraries` (`runtime libraries` is a renewed name for corefx)